src/bin/ipc_latency.rs

```diff
@@ -88,18 +134,45 @@ macro_rules! netlink_bench {
                 .output()
                 .expect("make failed to start");
 
-            let (tx, rx) = mpsc::channel::<Vec<Duration>>();
+            let (tx, rx) = mpsc::channel::<Vec<(Duration, Duration, Duration)>>();
 
             // listen
             let c1 = thread::spawn(move || {
                 let mut buf = [0u8; 1024];
-                let nl = portus::ipc::netlink::Socket::<$mode>::new()
-                    .map(|sk| {
-                        Backend::new(sk, Arc::new(atomic::AtomicBool::new(true)), &mut buf[..])
-                    })
+                // What does the backend represent?
+                let mut nl = portus::ipc::netlink::Socket::<$mode>::new()
+                    .map(|sk| Backend::new(sk, Arc::new(atomic::AtomicBool::new(true)), &mut buf[..]))
                     .expect("nl ipc initialization");
                 tx.send(vec![]).expect("ok to insmod");
-                tx.send(bench(&nl.sender(), nl, iter)).expect("report rtts");
+                // this is what I added
+                nl.next().expect("receive echo");
+
+                let sender = nl.sender();
+                let res  = (0..iter)
+                .map(|_| {
+
+                    let portus_send_time= time::get_time(); 
+                    let mut portus_recv_time = time::get_time();
+                    let mut kern_recv_time = time::get_time();
+                    let mut kern_send_time = time::get_time();
```

Comment:
instead of initializing `mut` variables, have them be returned from the `if let` below - `mut` variables are harder to reason about and all the `time::get_time()` which are then overwritten is hard to parse.

Something like:
```rust
let (portus_recv_time, kern_recv_time, kern_send_time) = if let portus::serialize::Msg::Other(raw) = nl.next().expect("blah") {
  let portus_rt = get_time();
  let let kern_recv_msg = DetTimeMsg::from_raw_msg(raw).expect("get time from raw");
  (portus_rt, kern_recv_msg.0, kern_recv_msg.1)
} else {
  panic!(..)
};
```

Reply:
Yes. This makes a lot of sense. I'll change it.